tests/_runner.lua

```diff
@@ -136,18 +138,26 @@ function runner.run_steps(steps, options)
                 -- All steps finished, we are done.
                 runner.done()
             end
-        elseif result == false then
-            runner.done("Step "..step_as_string.." failed (returned false).")
         else
-            -- No result yet, run this step again.
-            wait = wait-1
-            if wait > 0 then
-                t.timeout = 0.1
-                t:again()
+            -- Append filename/lnum of failed step function.
+            local step_info = debug.getinfo(step_func)
+            local step_loc = string.format("%s:%d", step_info["short_src"], step_info["linedefined"])
+            step_as_string = step_as_string .. " ("..step_loc..")"
+
+            if result == false then
+                runner.done("Step "..step_as_string.." failed (returned false).")
             else
-                runner.done("timeout waiting for signal in step "
-                            ..step_as_string..".")
-                t:stop()
+                -- No result yet, run this step again.
+                wait = wait-1
+                if wait > 0 then
+                    t.timeout = 0.1
+                    t:again()
+                else
+                    require("gears.debug").dump(debug.getinfo(step_func))
```

Comment:
Is this addition a left-over from development?

Reply:
Yeah.